wasm.md

```diff
@@ -583,13 +583,19 @@ A type use is a reference to a type definition.
 It may optionally be augmented by explicit inlined parameter and result declarations.
 A type use should start with `'(' 'type' x:typeidx ')'` followed by a group of inlined parameter or result declarations.
 
-# TODO: move to wasm-text
+# TODO: Remove the middle case (single `(type X)` without declaration), and move to wasm-text.
+
 ```k
     syntax TypeUse ::= TypeDecls
                      | "(type" Index ")"           [prefer]
                      | "(type" Index ")" TypeDecls
  // ----------------------------------------------
 
+    syntax Index ::= #unsafeExtractIndex ( TypeUse ) [function]
+ // -----------------------------------------------------------
+    rule #unsafeExtractIndex((type X)            ) => X
+    rule #unsafeExtractIndex((type X) _:TypeDecls) => X
```

Comment:
what's unsafe about this?

Reply:
It may be a `TypeDecls`.

Reply:
Then I would add a subsort:

```
syntax TypeUse ::= TypeDecls | TypeSomethingElse

syntax TypeSomethingElse ::= "(type" Index ")" | "(type" Index ")" TypeDecls
```

And make it over sort `TypeSomethingElse` (with a better name).

I also notice this function is only used in one location, so maybe move it closer to that use (whether adding the subsort or not). And at the use sight, assert on the LHS of the rule that the input is the correct sort (either by using the subsort, or by adding `notBool isTypeDecls(TUSE)`.

Reply:
Have a look at how imports work in commit 9a028ef. I think this is cleaner, more in line with how we want the abstract format to look (with records and exact types whenever possible). So the "unsafe" part would now instead get caught earlier, in the translation from the text format.